pdns/ixfr.cc

```diff
@@ -213,6 +217,12 @@ vector<pair<vector<DNSRecord>, vector<DNSRecord> > > getIXFRDeltas(const ComboAd
           return ret;
         }
         masterSOA = sr;
+      } else {
+        // we hit the last SOA record
+        // ixfr is considered to be done if we hit the last SOA record twice
+        if (r.first.d_type == QType::SOA && masterSOA->d_st.serial == getRR<SOARecordContent>(r.first)->d_st.serial) {
```

Comment:
Please check that `getRR<SOARecordContent>(r.first)` is not `nullptr` before dereferencing it. It might happen is the class is not `IN` for example.

Reply:
Thanks rgacogne for reviewing. Good sense of that. I've updated the code accordingly.